pkg/certificate/providers/mrc.go

```diff
@@ -0,0 +1,118 @@
+package providers
```

Comment:
thoughts on putting this in `pkg/k8s`, since this is a k8s specific implementation of the MRCClient?

Reply:
The `MRCProviderGenerator` field/embedding would make that an import cycle. `pkg/certificate/providers` imports `pkg/k8s` which would import `pkg/certificate/providers` where `MRCProviderGenerator is stored